{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Table of Contents\n",
    " <p><div class=\"lev1\"><a href=\"#Task-1.-Compiling-Ebola-Data\"><span class=\"toc-item-num\">Task 1.&nbsp;&nbsp;</span>Compiling Ebola Data</a></div>\n",
    " <div class=\"lev1\"><a href=\"#Task-2.-RNA-Sequences\"><span class=\"toc-item-num\">Task 2.&nbsp;&nbsp;</span>RNA Sequences</a></div>\n",
    " <div class=\"lev1\"><a href=\"#Task-3.-Class-War-in-Titanic\"><span class=\"toc-item-num\">Task 3.&nbsp;&nbsp;</span>Class War in Titanic</a></div></p>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 552,
   "metadata": {},
   "outputs": [],
   "source": [
    "DATA_FOLDER = 'Data' # Use the data folder provided in Tutorial 02 - Intro to Pandas."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Task 1. Compiling Ebola Data\n",
    "\n",
    "The `DATA_FOLDER/ebola` folder contains summarized reports of Ebola cases from three countries (Guinea, Liberia and Sierra Leone) during the recent outbreak of the disease in West Africa. For each country, there are daily reports that contain various information about the outbreak in several cities in each country.\n",
    "\n",
    "Use pandas to import these data files into a single `Dataframe`.\n",
    "Using this `DataFrame`, calculate for *each country*, the *daily average per month* of *new cases* and *deaths*.\n",
    "Make sure you handle all the different expressions for *new cases* and *deaths* that are used in the reports."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 553,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import glob\n",
    "from datetime import date,time,datetime\n",
    "import time\n",
    "import xlrd\n",
    "# Write your answer here"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 554,
   "metadata": {},
   "outputs": [],
   "source": [
    "#input the data of guinea\n",
    "path = r'Data\\ebola\\guinea_data'\n",
    "allFiles = glob.glob(path + \"/*.csv\")\n",
    "frame1 = pd.DataFrame()\n",
    "list_ = []\n",
    "for file_ in allFiles:\n",
    "    df = pd.read_csv(file_,index_col=None,header=0)\n",
    "    list_.append(df)\n",
    "frame1 = pd.concat(list_)\n",
    "\n",
    "path = r'Data\\ebola\\liberia_data'\n",
    "allFiles = glob.glob(path + \"/*.csv\")\n",
    "frame2 = pd.DataFrame()\n",
    "list_ = []\n",
    "for file_ in allFiles:\n",
    "    df = pd.read_csv(file_,index_col=None,header=0)\n",
    "    list_.append(df)\n",
    "frame2 = pd.concat(list_)\n",
    "\n",
    "path = r'Data\\ebola\\sl_data'\n",
    "allFiles = glob.glob(path + \"/*.csv\")\n",
    "frame3 = pd.DataFrame()\n",
    "list_ = []\n",
    "for file_ in allFiles:\n",
    "    df = pd.read_csv(file_,index_col=None,header=0)\n",
    "    list_.append(df)\n",
    "frame3 = pd.concat(list_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 555,
   "metadata": {},
   "outputs": [],
   "source": [
    "#rename the \"Description\" colunmn in order to make the same columns for the three countries\n",
    "frame1 = frame1.rename(index=str, columns={\"Description\": \"Variable\"})\n",
    "frame3 = frame3.rename(index=str, columns={\"variable\": \"Variable\",\"date\": \"Date\"})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 556,
   "metadata": {},
   "outputs": [],
   "source": [
    "#add the tap of country\n",
    "frame1['Country'] = 'guinea'\n",
    "frame2['Country'] = 'liberia'\n",
    "frame3['Country'] = 'sl'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 557,
   "metadata": {},
   "outputs": [],
   "source": [
    "#in order to combine the datas of three countries, we make them the same 'Variable'\n",
    "frame1.rename(index=str, columns={\"Totals\": \"National\"},inplace=True)\n",
    "frame2.replace('Total confirmed cases','Total cases of confirmed',inplace=True)\n",
    "frame2.replace('Total probable cases','Total cases of probables',inplace=True)\n",
    "frame2.replace('Total suspected cases','Total cases of suspects',inplace=True)\n",
    "frame2.replace('Total death/s in confirmed cases','Total deaths of confirmed',inplace=True)\n",
    "frame2.replace('Total death/s in probable cases','Total deaths of probables',inplace=True)\n",
    "frame2.replace('Total death/s in suspected cases','Total deaths of suspects',inplace=True)\n",
    "\n",
    "frame3.replace('cum_confirmed','Total cases of confirmed',inplace=True)\n",
    "frame3.replace('cum_probable','Total cases of probables',inplace=True)\n",
    "frame3.replace('cum_suspected','Total cases of suspects',inplace=True)\n",
    "frame3.replace('death_confirmed','Total deaths of confirmed',inplace=True)\n",
    "frame3.replace('death_probable','Total deaths of probables',inplace=True)\n",
    "frame3.replace('death_suspected','Total deaths of suspects',inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 558,
   "metadata": {},
   "outputs": [],
   "source": [
    "#since we only need to calculate data for each countriy, so we chose the data we want.\n",
    "frame1 = frame1[['Date','Country','Variable','National']]\n",
    "frame2 = frame2[['Date','Country','Variable','National']]\n",
    "frame3 = frame3[['Date','Country','Variable','National']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 559,
   "metadata": {},
   "outputs": [],
   "source": [
    "#merge data of three countries\n",
    "DF = pd.merge(frame1,frame2,how='outer')\n",
    "df = pd.merge(DF,frame3,how='outer')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 560,
   "metadata": {},
   "outputs": [],
   "source": [
    "#transfer Date data from str to datetime \n",
    "df['Date'] = pd.to_datetime(df.Date)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 561,
   "metadata": {},
   "outputs": [],
   "source": [
    "#get the data about 'Total cases' and 'Total deaths'. In my opinion, the other data such as 'New cases of suspects' are not necessary data for \n",
    "#us to calculate daily average per month of new cases and deaths, since data with 'Total' can truely reflect the situation while the 'New' can\n",
    "#only reflect specific days' situation rather than per month.\n",
    "sub1 = df[[name.startswith('Total cases of') for name in df.Variable]]\n",
    "sub2 = df[[name.startswith('Total deaths of') for name in df.Variable]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 562,
   "metadata": {},
   "outputs": [],
   "source": [
    "# merge the dataframe into a single dataframe\n",
    "Data1 = pd.merge(sub1,sub2,how='outer')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 563,
   "metadata": {},
   "outputs": [],
   "source": [
    "Data= Data1.set_index(['Country','Variable','Date']).sort_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 564,
   "metadata": {},
   "outputs": [],
   "source": [
    "#delete the NAN \n",
    "Data = Data.dropna()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 565,
   "metadata": {},
   "outputs": [],
   "source": [
    "#some value of 'National' are strings like 1,276, we should transfer them to 1276\n",
    "Data.loc['sl','Total cases of confirmed','2014--09-06']=1276\n",
    "Data.loc['sl','Total cases of confirmed','2014--09-17']=1585\n",
    "Data.loc['sl','Total cases of confirmed','2014--09-18']=1618\n",
    "Data.loc['sl','Total cases of confirmed','2014--09-20']=1696\n",
    "Data.loc['sl','Total cases of confirmed','2014--09-027']=2000"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 567,
   "metadata": {},
   "outputs": [],
   "source": [
    "Str = ['guinea','liberia','sl']\n",
    "String = ['Total cases of confirmed','Total cases of probables','Total cases of suspects','Total deaths of confirmed','Total deaths of probables','Total deaths of suspects']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 591,
   "metadata": {},
   "outputs": [],
   "source": [
    "Ave = np.zeros([3,6])\n",
    "for i in range(0,3):\n",
    "    for j in range(0,6):\n",
    "        Lenth = Data.loc[Str[i],String[j]].shape[0]\n",
    "        harf_lenth = int(Lenth/2)\n",
    "        sub = Data.loc[Str[i],String[j]] #choose the data we wanted\n",
    "        sub['National'] = pd.to_numeric(sub.National)#since values of 'National' are strings, we should transfer them into int or float\n",
    "        sub1= sub[:harf_lenth]#we divide the data into two groups so that we can make full use of each data\n",
    "        sub2 =sub[harf_lenth:2*harf_lenth]\n",
    "        sub1= sub1.reset_index()\n",
    "        sub2= sub2.reset_index()\n",
    "        diff = sub2 - sub1 #For example we have 48 data for National adn we use 25th data to subtranct 1st data and then Divided by the time difference\n",
    "        ave = np.zeros([harf_lenth,1])\n",
    "        for t in range (0,harf_lenth):\n",
    "            ave[t] = diff.National[t]/diff.Date[t].days\n",
    "        Ave[i][j] = sum(ave)/(harf_lenth)        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 592,
   "metadata": {},
   "outputs": [],
   "source": [
    "AVE = pd.DataFrame(Ave)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 600,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Total cases of confirmed</th>\n",
       "      <th>Total cases of probables</th>\n",
       "      <th>Total cases of suspects</th>\n",
       "      <th>Total deaths of confirmed</th>\n",
       "      <th>Total deaths of probables</th>\n",
       "      <th>Total deaths of suspects</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>guinea</th>\n",
       "      <td>12.316357</td>\n",
       "      <td>0.854977</td>\n",
       "      <td>-0.080584</td>\n",
       "      <td>6.980914</td>\n",
       "      <td>2.496392</td>\n",
       "      <td>1.073234</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>liberia</th>\n",
       "      <td>19.585189</td>\n",
       "      <td>15.020251</td>\n",
       "      <td>22.360379</td>\n",
       "      <td>10.798210</td>\n",
       "      <td>7.976506</td>\n",
       "      <td>5.558687</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sl</th>\n",
       "      <td>49.146334</td>\n",
       "      <td>0.451809</td>\n",
       "      <td>10.297422</td>\n",
       "      <td>12.371717</td>\n",
       "      <td>2.711003</td>\n",
       "      <td>2.176776</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         Total cases of confirmed  Total cases of probables  \\\n",
       "guinea                  12.316357                  0.854977   \n",
       "liberia                 19.585189                 15.020251   \n",
       "sl                      49.146334                  0.451809   \n",
       "\n",
       "         Total cases of suspects  Total deaths of confirmed  \\\n",
       "guinea                 -0.080584                   6.980914   \n",
       "liberia                22.360379                  10.798210   \n",
       "sl                     10.297422                  12.371717   \n",
       "\n",
       "         Total deaths of probables  Total deaths of suspects  \n",
       "guinea                    2.496392                  1.073234  \n",
       "liberia                   7.976506                  5.558687  \n",
       "sl                        2.711003                  2.176776  "
      ]
     },
     "execution_count": 600,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "AVE.columns = String\n",
    "AVE.index=[Str]\n",
    "AVE"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Task 2. RNA Sequences\n",
    "\n",
    "In the `DATA_FOLDER/microbiome` subdirectory, there are 9 spreadsheets of microbiome data that was acquired from high-throughput RNA sequencing procedures, along with a 10<sup>th</sup> file that describes the content of each. \n",
    "\n",
    "Use pandas to import the first 9 spreadsheets into a single `DataFrame`.\n",
    "Then, add the metadata information from the 10<sup>th</sup> spreadsheet as columns in the combined `DataFrame`.\n",
    "Make sure that the final `DataFrame` has a unique index and all the `NaN` values have been replaced by the tag `unknown`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 609,
   "metadata": {},
   "outputs": [],
   "source": [
    "#input data and set the colunms\n",
    "path = r'Data\\microbiome'\n",
    "allFiles = glob.glob(path + \"/*.xls\")\n",
    "dfs = pd.DataFrame()\n",
    "list_= []\n",
    "i = 1\n",
    "for file_ in allFiles[1:10]:\n",
    "    df= pd.read_excel(file_, 'Sheet 1', index_col=None, header=None)\n",
    "    String = 'MID'+str(i)\n",
    "    df['BARCODE'] =String\n",
    "    i = i + 1\n",
    "    dfs = pd.concat([dfs,df],axis=0)\n",
    "dfs.columns = ['Taxon','Count','BARCODE']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 610,
   "metadata": {},
   "outputs": [],
   "source": [
    "X_file = pd.ExcelFile(allFiles[0])\n",
    "Df=X_file.parse('Sheet1')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 605,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "#add the metadata information from the 10th spreadsheet as columns in the combined DataFrame\n",
    "Data = pd.merge(dfs,Df,on='BARCODE',right_index = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 611,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Taxon</th>\n",
       "      <th>Count</th>\n",
       "      <th>BARCODE</th>\n",
       "      <th>GROUP</th>\n",
       "      <th>SAMPLE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Archaea \"Crenarchaeota\" Thermoprotei Desulfuro...</td>\n",
       "      <td>7</td>\n",
       "      <td>MID1</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Archaea \"Crenarchaeota\" Thermoprotei Desulfuro...</td>\n",
       "      <td>2</td>\n",
       "      <td>MID1</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Archaea \"Crenarchaeota\" Thermoprotei Sulfoloba...</td>\n",
       "      <td>3</td>\n",
       "      <td>MID1</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Archaea \"Crenarchaeota\" Thermoprotei Thermopro...</td>\n",
       "      <td>3</td>\n",
       "      <td>MID1</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Archaea \"Euryarchaeota\" \"Methanomicrobia\" Meth...</td>\n",
       "      <td>7</td>\n",
       "      <td>MID1</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                               Taxon  Count BARCODE  \\\n",
       "0  Archaea \"Crenarchaeota\" Thermoprotei Desulfuro...      7    MID1   \n",
       "1  Archaea \"Crenarchaeota\" Thermoprotei Desulfuro...      2    MID1   \n",
       "2  Archaea \"Crenarchaeota\" Thermoprotei Sulfoloba...      3    MID1   \n",
       "3  Archaea \"Crenarchaeota\" Thermoprotei Thermopro...      3    MID1   \n",
       "4  Archaea \"Euryarchaeota\" \"Methanomicrobia\" Meth...      7    MID1   \n",
       "\n",
       "                GROUP   SAMPLE  \n",
       "0  EXTRACTION CONTROL  unknown  \n",
       "1  EXTRACTION CONTROL  unknown  \n",
       "2  EXTRACTION CONTROL  unknown  \n",
       "3  EXTRACTION CONTROL  unknown  \n",
       "4  EXTRACTION CONTROL  unknown  "
      ]
     },
     "execution_count": 611,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#all the NaN values have been replaced by the tag unknown\n",
    "Data.fillna('unknown')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 608,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>Count</th>\n",
       "      <th>BARCODE</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Taxon</th>\n",
       "      <th>GROUP</th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"3\" valign=\"top\">Archaea \"Crenarchaeota\" Thermoprotei Acidilobales Acidilobaceae Acidilobus</th>\n",
       "      <th>Control 1</th>\n",
       "      <th>tissue</th>\n",
       "      <td>1</td>\n",
       "      <td>MID3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Control 2</th>\n",
       "      <th>tissue</th>\n",
       "      <td>5</td>\n",
       "      <td>MID5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NEC 1</th>\n",
       "      <th>tissue</th>\n",
       "      <td>2</td>\n",
       "      <td>MID2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">Archaea \"Crenarchaeota\" Thermoprotei Acidilobales Caldisphaeraceae Caldisphaera</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">Control 1</th>\n",
       "      <th>stool</th>\n",
       "      <td>1</td>\n",
       "      <td>MID7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tissue</th>\n",
       "      <td>15</td>\n",
       "      <td>MID3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">Control 2</th>\n",
       "      <th>stool</th>\n",
       "      <td>1</td>\n",
       "      <td>MID9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tissue</th>\n",
       "      <td>26</td>\n",
       "      <td>MID5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NEC 1</th>\n",
       "      <th>tissue</th>\n",
       "      <td>14</td>\n",
       "      <td>MID2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"8\" valign=\"top\">Archaea \"Crenarchaeota\" Thermoprotei Desulfurococcales Desulfurococcaceae Ignisphaera</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">Control 1</th>\n",
       "      <th>stool</th>\n",
       "      <td>8</td>\n",
       "      <td>MID7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tissue</th>\n",
       "      <td>14</td>\n",
       "      <td>MID3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">Control 2</th>\n",
       "      <th>stool</th>\n",
       "      <td>16</td>\n",
       "      <td>MID9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tissue</th>\n",
       "      <td>28</td>\n",
       "      <td>MID5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EXTRACTION CONTROL</th>\n",
       "      <th>NaN</th>\n",
       "      <td>7</td>\n",
       "      <td>MID1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">NEC 1</th>\n",
       "      <th>stool</th>\n",
       "      <td>7</td>\n",
       "      <td>MID6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tissue</th>\n",
       "      <td>23</td>\n",
       "      <td>MID2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NEC 2</th>\n",
       "      <th>tissue</th>\n",
       "      <td>2</td>\n",
       "      <td>MID4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Archaea \"Crenarchaeota\" Thermoprotei Desulfurococcales Desulfurococcaceae Stetteria</th>\n",
       "      <th>Control 2</th>\n",
       "      <th>tissue</th>\n",
       "      <td>1</td>\n",
       "      <td>MID5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"6\" valign=\"top\">Archaea \"Crenarchaeota\" Thermoprotei Desulfurococcales Desulfurococcaceae Sulfophobococcus</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">Control 1</th>\n",
       "      <th>stool</th>\n",
       "      <td>2</td>\n",
       "      <td>MID7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tissue</th>\n",
       "      <td>4</td>\n",
       "      <td>MID3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">Control 2</th>\n",
       "      <th>stool</th>\n",
       "      <td>2</td>\n",
       "      <td>MID9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tissue</th>\n",
       "      <td>5</td>\n",
       "      <td>MID5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">NEC 1</th>\n",
       "      <th>stool</th>\n",
       "      <td>1</td>\n",
       "      <td>MID6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tissue</th>\n",
       "      <td>1</td>\n",
       "      <td>MID2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Archaea \"Crenarchaeota\" Thermoprotei Desulfurococcales Desulfurococcaceae Thermodiscus</th>\n",
       "      <th>Control 1</th>\n",
       "      <th>tissue</th>\n",
       "      <td>1</td>\n",
       "      <td>MID3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">Archaea \"Crenarchaeota\" Thermoprotei Desulfurococcales Desulfurococcaceae Thermosphaera</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">Control 1</th>\n",
       "      <th>stool</th>\n",
       "      <td>1</td>\n",
       "      <td>MID7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tissue</th>\n",
       "      <td>1</td>\n",
       "      <td>MID3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Control 2</th>\n",
       "      <th>tissue</th>\n",
       "      <td>2</td>\n",
       "      <td>MID5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NEC 1</th>\n",
       "      <th>tissue</th>\n",
       "      <td>2</td>\n",
       "      <td>MID2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Archaea \"Crenarchaeota\" Thermoprotei Desulfurococcales Pyrodictiaceae Hyperthermus</th>\n",
       "      <th>NEC 1</th>\n",
       "      <th>tissue</th>\n",
       "      <td>1</td>\n",
       "      <td>MID2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Archaea \"Crenarchaeota\" Thermoprotei Desulfurococcales Pyrodictiaceae Pyrodictium</th>\n",
       "      <th>Control 1</th>\n",
       "      <th>stool</th>\n",
       "      <td>1</td>\n",
       "      <td>MID7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteria \"Verrucomicrobia\" Subdivision3   Subdivision3_genera_incertae_sedis</th>\n",
       "      <th>Control 1</th>\n",
       "      <th>stool</th>\n",
       "      <td>4</td>\n",
       "      <td>MID7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"3\" valign=\"top\">Bacteria \"Verrucomicrobia\" Verrucomicrobiae Verrucomicrobiales Verrucomicrobiaceae Persicirhabdus</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">Control 1</th>\n",
       "      <th>stool</th>\n",
       "      <td>9</td>\n",
       "      <td>MID7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tissue</th>\n",
       "      <td>2</td>\n",
       "      <td>MID3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NEC 1</th>\n",
       "      <th>stool</th>\n",
       "      <td>1</td>\n",
       "      <td>MID6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"3\" valign=\"top\">Bacteria \"Verrucomicrobia\" Verrucomicrobiae Verrucomicrobiales Verrucomicrobiaceae Verrucomicrobium</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">Control 1</th>\n",
       "      <th>stool</th>\n",
       "      <td>39</td>\n",
       "      <td>MID7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tissue</th>\n",
       "      <td>1</td>\n",
       "      <td>MID3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NEC 1</th>\n",
       "      <th>stool</th>\n",
       "      <td>1</td>\n",
       "      <td>MID6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">Bacteria Cyanobacteria Cyanobacteria  Chloroplast Bangiophyceae</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">Control 1</th>\n",
       "      <th>stool</th>\n",
       "      <td>1</td>\n",
       "      <td>MID7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tissue</th>\n",
       "      <td>1</td>\n",
       "      <td>MID3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Control 2</th>\n",
       "      <th>tissue</th>\n",
       "      <td>2</td>\n",
       "      <td>MID5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EXTRACTION CONTROL</th>\n",
       "      <th>NaN</th>\n",
       "      <td>2</td>\n",
       "      <td>MID1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NEC 2</th>\n",
       "      <th>stool</th>\n",
       "      <td>1</td>\n",
       "      <td>MID8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"8\" valign=\"top\">Bacteria Cyanobacteria Cyanobacteria  Chloroplast Chlorarachniophyceae</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">Control 1</th>\n",
       "      <th>stool</th>\n",
       "      <td>2</td>\n",
       "      <td>MID7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tissue</th>\n",
       "      <td>4</td>\n",
       "      <td>MID3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">Control 2</th>\n",
       "      <th>stool</th>\n",
       "      <td>1</td>\n",
       "      <td>MID9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tissue</th>\n",
       "      <td>5</td>\n",
       "      <td>MID5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EXTRACTION CONTROL</th>\n",
       "      <th>NaN</th>\n",
       "      <td>85</td>\n",
       "      <td>MID1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NEC 1</th>\n",
       "      <th>tissue</th>\n",
       "      <td>1</td>\n",
       "      <td>MID2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">NEC 2</th>\n",
       "      <th>stool</th>\n",
       "      <td>4</td>\n",
       "      <td>MID8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tissue</th>\n",
       "      <td>6</td>\n",
       "      <td>MID4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"6\" valign=\"top\">Bacteria Cyanobacteria Cyanobacteria  Chloroplast Streptophyta</th>\n",
       "      <th>Control 1</th>\n",
       "      <th>tissue</th>\n",
       "      <td>54</td>\n",
       "      <td>MID3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">Control 2</th>\n",
       "      <th>stool</th>\n",
       "      <td>10</td>\n",
       "      <td>MID9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tissue</th>\n",
       "      <td>59</td>\n",
       "      <td>MID5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EXTRACTION CONTROL</th>\n",
       "      <th>NaN</th>\n",
       "      <td>1388</td>\n",
       "      <td>MID1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NEC 1</th>\n",
       "      <th>tissue</th>\n",
       "      <td>2</td>\n",
       "      <td>MID2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NEC 2</th>\n",
       "      <th>tissue</th>\n",
       "      <td>44</td>\n",
       "      <td>MID4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">Bacteria OD1    OD1_genera_incertae_sedis</th>\n",
       "      <th>Control 1</th>\n",
       "      <th>tissue</th>\n",
       "      <td>1</td>\n",
       "      <td>MID3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Control 2</th>\n",
       "      <th>tissue</th>\n",
       "      <td>4</td>\n",
       "      <td>MID5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteria OP11    OP11_genera_incertae_sedis</th>\n",
       "      <th>Control 1</th>\n",
       "      <th>tissue</th>\n",
       "      <td>26</td>\n",
       "      <td>MID3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteria TM7    TM7_genera_incertae_sedis</th>\n",
       "      <th>NEC 1</th>\n",
       "      <th>tissue</th>\n",
       "      <td>2</td>\n",
       "      <td>MID2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2396 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                              Count  \\\n",
       "Taxon                                              GROUP              SAMPLE          \n",
       "Archaea \"Crenarchaeota\" Thermoprotei Acidilobal... Control 1          tissue      1   \n",
       "                                                   Control 2          tissue      5   \n",
       "                                                   NEC 1              tissue      2   \n",
       "Archaea \"Crenarchaeota\" Thermoprotei Acidilobal... Control 1          stool       1   \n",
       "                                                                      tissue     15   \n",
       "                                                   Control 2          stool       1   \n",
       "                                                                      tissue     26   \n",
       "                                                   NEC 1              tissue     14   \n",
       "Archaea \"Crenarchaeota\" Thermoprotei Desulfuroc... Control 1          stool       8   \n",
       "                                                                      tissue     14   \n",
       "                                                   Control 2          stool      16   \n",
       "                                                                      tissue     28   \n",
       "                                                   EXTRACTION CONTROL NaN         7   \n",
       "                                                   NEC 1              stool       7   \n",
       "                                                                      tissue     23   \n",
       "                                                   NEC 2              tissue      2   \n",
       "Archaea \"Crenarchaeota\" Thermoprotei Desulfuroc... Control 2          tissue      1   \n",
       "Archaea \"Crenarchaeota\" Thermoprotei Desulfuroc... Control 1          stool       2   \n",
       "                                                                      tissue      4   \n",
       "                                                   Control 2          stool       2   \n",
       "                                                                      tissue      5   \n",
       "                                                   NEC 1              stool       1   \n",
       "                                                                      tissue      1   \n",
       "Archaea \"Crenarchaeota\" Thermoprotei Desulfuroc... Control 1          tissue      1   \n",
       "Archaea \"Crenarchaeota\" Thermoprotei Desulfuroc... Control 1          stool       1   \n",
       "                                                                      tissue      1   \n",
       "                                                   Control 2          tissue      2   \n",
       "                                                   NEC 1              tissue      2   \n",
       "Archaea \"Crenarchaeota\" Thermoprotei Desulfuroc... NEC 1              tissue      1   \n",
       "Archaea \"Crenarchaeota\" Thermoprotei Desulfuroc... Control 1          stool       1   \n",
       "...                                                                             ...   \n",
       "Bacteria \"Verrucomicrobia\" Subdivision3   Subdi... Control 1          stool       4   \n",
       "Bacteria \"Verrucomicrobia\" Verrucomicrobiae Ver... Control 1          stool       9   \n",
       "                                                                      tissue      2   \n",
       "                                                   NEC 1              stool       1   \n",
       "Bacteria \"Verrucomicrobia\" Verrucomicrobiae Ver... Control 1          stool      39   \n",
       "                                                                      tissue      1   \n",
       "                                                   NEC 1              stool       1   \n",
       "Bacteria Cyanobacteria Cyanobacteria  Chloropla... Control 1          stool       1   \n",
       "                                                                      tissue      1   \n",
       "                                                   Control 2          tissue      2   \n",
       "                                                   EXTRACTION CONTROL NaN         2   \n",
       "                                                   NEC 2              stool       1   \n",
       "Bacteria Cyanobacteria Cyanobacteria  Chloropla... Control 1          stool       2   \n",
       "                                                                      tissue      4   \n",
       "                                                   Control 2          stool       1   \n",
       "                                                                      tissue      5   \n",
       "                                                   EXTRACTION CONTROL NaN        85   \n",
       "                                                   NEC 1              tissue      1   \n",
       "                                                   NEC 2              stool       4   \n",
       "                                                                      tissue      6   \n",
       "Bacteria Cyanobacteria Cyanobacteria  Chloropla... Control 1          tissue     54   \n",
       "                                                   Control 2          stool      10   \n",
       "                                                                      tissue     59   \n",
       "                                                   EXTRACTION CONTROL NaN      1388   \n",
       "                                                   NEC 1              tissue      2   \n",
       "                                                   NEC 2              tissue     44   \n",
       "Bacteria OD1    OD1_genera_incertae_sedis          Control 1          tissue      1   \n",
       "                                                   Control 2          tissue      4   \n",
       "Bacteria OP11    OP11_genera_incertae_sedis        Control 1          tissue     26   \n",
       "Bacteria TM7    TM7_genera_incertae_sedis          NEC 1              tissue      2   \n",
       "\n",
       "                                                                             BARCODE  \n",
       "Taxon                                              GROUP              SAMPLE          \n",
       "Archaea \"Crenarchaeota\" Thermoprotei Acidilobal... Control 1          tissue    MID3  \n",
       "                                                   Control 2          tissue    MID5  \n",
       "                                                   NEC 1              tissue    MID2  \n",
       "Archaea \"Crenarchaeota\" Thermoprotei Acidilobal... Control 1          stool     MID7  \n",
       "                                                                      tissue    MID3  \n",
       "                                                   Control 2          stool     MID9  \n",
       "                                                                      tissue    MID5  \n",
       "                                                   NEC 1              tissue    MID2  \n",
       "Archaea \"Crenarchaeota\" Thermoprotei Desulfuroc... Control 1          stool     MID7  \n",
       "                                                                      tissue    MID3  \n",
       "                                                   Control 2          stool     MID9  \n",
       "                                                                      tissue    MID5  \n",
       "                                                   EXTRACTION CONTROL NaN       MID1  \n",
       "                                                   NEC 1              stool     MID6  \n",
       "                                                                      tissue    MID2  \n",
       "                                                   NEC 2              tissue    MID4  \n",
       "Archaea \"Crenarchaeota\" Thermoprotei Desulfuroc... Control 2          tissue    MID5  \n",
       "Archaea \"Crenarchaeota\" Thermoprotei Desulfuroc... Control 1          stool     MID7  \n",
       "                                                                      tissue    MID3  \n",
       "                                                   Control 2          stool     MID9  \n",
       "                                                                      tissue    MID5  \n",
       "                                                   NEC 1              stool     MID6  \n",
       "                                                                      tissue    MID2  \n",
       "Archaea \"Crenarchaeota\" Thermoprotei Desulfuroc... Control 1          tissue    MID3  \n",
       "Archaea \"Crenarchaeota\" Thermoprotei Desulfuroc... Control 1          stool     MID7  \n",
       "                                                                      tissue    MID3  \n",
       "                                                   Control 2          tissue    MID5  \n",
       "                                                   NEC 1              tissue    MID2  \n",
       "Archaea \"Crenarchaeota\" Thermoprotei Desulfuroc... NEC 1              tissue    MID2  \n",
       "Archaea \"Crenarchaeota\" Thermoprotei Desulfuroc... Control 1          stool     MID7  \n",
       "...                                                                              ...  \n",
       "Bacteria \"Verrucomicrobia\" Subdivision3   Subdi... Control 1          stool     MID7  \n",
       "Bacteria \"Verrucomicrobia\" Verrucomicrobiae Ver... Control 1          stool     MID7  \n",
       "                                                                      tissue    MID3  \n",
       "                                                   NEC 1              stool     MID6  \n",
       "Bacteria \"Verrucomicrobia\" Verrucomicrobiae Ver... Control 1          stool     MID7  \n",
       "                                                                      tissue    MID3  \n",
       "                                                   NEC 1              stool     MID6  \n",
       "Bacteria Cyanobacteria Cyanobacteria  Chloropla... Control 1          stool     MID7  \n",
       "                                                                      tissue    MID3  \n",
       "                                                   Control 2          tissue    MID5  \n",
       "                                                   EXTRACTION CONTROL NaN       MID1  \n",
       "                                                   NEC 2              stool     MID8  \n",
       "Bacteria Cyanobacteria Cyanobacteria  Chloropla... Control 1          stool     MID7  \n",
       "                                                                      tissue    MID3  \n",
       "                                                   Control 2          stool     MID9  \n",
       "                                                                      tissue    MID5  \n",
       "                                                   EXTRACTION CONTROL NaN       MID1  \n",
       "                                                   NEC 1              tissue    MID2  \n",
       "                                                   NEC 2              stool     MID8  \n",
       "                                                                      tissue    MID4  \n",
       "Bacteria Cyanobacteria Cyanobacteria  Chloropla... Control 1          tissue    MID3  \n",
       "                                                   Control 2          stool     MID9  \n",
       "                                                                      tissue    MID5  \n",
       "                                                   EXTRACTION CONTROL NaN       MID1  \n",
       "                                                   NEC 1              tissue    MID2  \n",
       "                                                   NEC 2              tissue    MID4  \n",
       "Bacteria OD1    OD1_genera_incertae_sedis          Control 1          tissue    MID3  \n",
       "                                                   Control 2          tissue    MID5  \n",
       "Bacteria OP11    OP11_genera_incertae_sedis        Control 1          tissue    MID3  \n",
       "Bacteria TM7    TM7_genera_incertae_sedis          NEC 1              tissue    MID2  \n",
       "\n",
       "[2396 rows x 2 columns]"
      ]
     },
     "execution_count": 608,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Data.set_index(['Taxon','GROUP','SAMPLE']).sort_index()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Task 3. Class War in Titanic\n",
    "\n",
    "Use pandas to import the data file `Data/titanic.xls`. It contains data on all the passengers that travelled on the Titanic."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "from IPython.core.display import HTML\n",
    "HTML(filename=DATA_FOLDER+'/titanic.html')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "For each of the following questions state clearly your assumptions and discuss your findings:\n",
    "1. Describe the *type* and the *value range* of each attribute. Indicate and transform the attributes that can be `Categorical`. \n",
    "2. Plot histograms for the *travel class*, *embarkation port*, *sex* and *age* attributes. For the latter one, use *discrete decade intervals*. \n",
    "3. Calculate the proportion of passengers by *cabin floor*. Present your results in a *pie chart*.\n",
    "4. For each *travel class*, calculate the proportion of the passengers that survived. Present your results in *pie charts*.\n",
    "5. Calculate the proportion of the passengers that survived by *travel class* and *sex*. Present your results in *a single histogram*.\n",
    "6. Create 2 equally populated *age categories* and calculate survival proportions by *age category*, *travel class* and *sex*. Present your results in a `DataFrame` with unique index."
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
